service.js

```diff
@@ -48,21 +48,38 @@ function getUrlMetadata(url) {
 }
 
 
-const app = express();
-app.use(bodyParser.json()); // for parsing application/json
-
-app.post('/', function(req, res) {
-  const promises = req.body.urls.map((url) => {
-    return getUrlMetadata(url);
-  });
+const server = new Hapi.Server();
+server.connection({
+  port: 7001
+});
 
-  Promise.all(promises).then((urlsData) => {
-    res.json({
-      error: '',
-      urls: buildObj(urlsData.map((urlData) => [urlData.url, urlData]))
-    });
-  });
+server.route({
+  method: 'POST',
+  path: '/',
+  config: {
+    validate: {
+      payload: {
+        urls: Joi.array().required()
+      }
+    }
+  },
```

Comment:
This `config.validate` section makes sure the user passed a required `urls` array in the POST payload.

### Good!

```sh
$ curl -XPOST http://localhost:7001/ -H "content-type: application/json" -d '{"urls": ["http://bing.com"]}' -i

HTTP/1.1 200 OK
content-type: application/json; charset=utf-8
cache-control: no-cache
content-length: 455
Date: Wed, 29 Jun 2016 23:01:35 GMT
Connection: keep-alive

{
  "error": "",
  "urls": {
    "http://bing.com": {
      "description": "Bing helps you turn information into action, making it faster and easier to go from searching to doing.",
      "icon_url": "http://www.bing.com/fd/s/a/hp/bing.svg",
      "image_url": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAEALAAAAAABAAEAAAIBTAA7",
      "title": "Bing",
      "url": "http://bing.com",
      "original_url": "http://bing.com",
      "provider_url": "http://bing.com",
      "favicon_url": "http://bing.com/favicon.ico"
    }
  }
}
```

### Bad!

```sh
$ curl -XPOST http://localhost:7001/ -H "content-type: application/json" -d '{}' -i

HTTP/1.1 400 Bad Request
content-type: application/json; charset=utf-8
cache-control: no-cache
content-length: 154
Date: Wed, 29 Jun 2016 23:03:12 GMT
Connection: keep-alive

{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "child \"urls\" fails because [\"urls\" is required]",
  "validation": {
    "source": "payload",
    "keys": [
      "urls"
    ]
  }
}
```